engine/movie/opening_cutscene.asm

```diff
@@ -0,0 +1,1235 @@
+	call InitSpriteAnimStruct
+	ret
+
+Intro_UpdateTilemapAndBGMap:
+; add new tiles to top as water scene scrolls up to surface
+	push hl
+	push de
+
+	ld a, [wIntroTilemapPointer + 0]
+	ld e, a
+	ld a, [wIntroTilemapPointer + 1]
+	ld d, a
+	ld hl, -$10
+	add hl, de
+	ld a, l
+	ld e, l
+	ld [wIntroTilemapPointer + 0], a
+	ld a, h
+	ld d, h
+	ld [wIntroTilemapPointer + 1], a
+
+	hlcoord 0, 0
+	ld c, BG_MAP_WIDTH / 2
+.loop
+	call Intro_Draw2x2Tiles
+	dec c
+	jr nz, .loop
+
+	ld a, [wIntroBGMapPointer + 0]
+	ld e, a
+	ld a, [wIntroBGMapPointer + 1]
+	ld d, a
+	ld hl, hCurSpriteYCoord
+	add hl, de
+	ld a, l
+	ld [wIntroBGMapPointer + 0], a
+	ld [wVBCopyDst], a
+	ld a, h
+	and %11111011
+	or %00001000
+	ld [wIntroBGMapPointer + 1], a
+	ld [wVBCopyDst + 1], a
+	ld a, LOW(wTileMap)
+	ld [wVBCopySrc], a
+	ld a, HIGH(wTileMap)
+	ld [wVBCopySrc + 1], a
+	ld a, 4
+	ld [wVBCopySize], a
+	ld hl, wIntroFrameCounter1
+	dec [hl]
+	pop de
+	pop hl
+	ret
+
+
+Intro_AnimateOceanWaves:
+; uses a 2bpp request to copy tile IDs to the BG map
+	ld hl, wIntroFrameCounter2
+	ld a, [hl]
+	and 3
+	cp 3
+	ret z
+
+	ld a, [hl]
+	and $30
+	swap a
+	ld l, a
+	ld h, 0
+rept 5
+	add hl, hl
+endr
+	ld de, .wave_tiles
+	add hl, de
+	ld a, l
+	ld [wVBCopySrc], a
+	ld a, h
+	ld [wVBCopySrc + 1], a
+	ld a, LOW(vBGMap0 + (15 * BG_MAP_WIDTH)) ; vBGMap0 row 15
+	ld [wVBCopyDst], a
+	ld a, HIGH(vBGMap0 + (15 * BG_MAP_WIDTH)) ; vBGMap0 row 15
+	ld [wVBCopyDst + 1], a
+	ld a, 2
+	ld [wVBCopySize], a
+	ret
+.wave_tiles:
+; Fill an entire bg map row with each frame
+rept 8
+	db $70, $71, $72, $73	; frame 1
+endr
+
+rept 8
+	db $74, $75, $76, $77	; frame 2
+endr
+
+rept 8
+	db $78, $79, $7a, $7b	; frame 3
+endr
+
+rept 8
+	db $7c, $7d, $7e, $7f	; frame 4
+endr
+
+Intro_InitSineLYOverrides:
+	ld bc, wLYOverrides2
+	ld a, SCREEN_HEIGHT_PX
+	ld de, BG_MAP_WIDTH * BG_MAP_HEIGHT
+.loop
+	push af
+	push de
+	callba BattleAnim_Sine_e
+	ld a, e
+	ld [bc], a
+	inc bc
+	pop de
+	inc e
+	pop af
+	dec a
+	jr nz, .loop
+	ret
+
+Intro_UpdateLYOverrides:
+	ld bc, wLYOverrides
+	ld e, $10
+
+.loop1
+	ldh a, [hSCY]
+	ld [bc], a
+	inc bc
+	dec e
+	jr nz, .loop1
+
+	ld hl, wLYOverrides2
+	ld de, wLYOverrides2 + 1
+	ld a, [hl]
+	push af
+	ld a, $80
+
+.loop2
+	push af
+	ld a, [de]
+	inc de
+	ld [hli], a
+	push hl
+	ld hl, hSCY
+	add [hl]
+	ld [bc], a
+	inc bc
+	pop hl
+	pop af
+	dec a
+	jr nz, .loop2
+
+	pop af
+	ld [hl], a
+	ret
+
+IntroScene6:
+; Set up grass cutscene (Pikachu/Jigglypuff)
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	call DisableLCD
+	ld b, SGB_GS_INTRO
+	ld c, 1
+	call GetSGBLayout
+	callab InitEffectObject
+	call Intro_ResetLYOverrides
+	ld hl, vChars2
+	ld de, IntroForestGFX
+	call Intro_Copy128Tiles
+	ld a, LOW(Intro_GrassMeta)
+	ld [wIntroTilesPointer + 0], a
+	ld a, HIGH(Intro_GrassMeta)
+	ld [wIntroTilesPointer + 1], a
+	ld hl, vBGMap0
+	ld a, l
+	ld [wIntroBGMapPointer + 0], a
+	ld a, h
+	ld [wIntroBGMapPointer + 1], a
+	ld de, Intro_GrassTilemap + 2 tiles
```

Comment:
This would be `Intro_GrassTilemap + $20`; and should the first $20 bytes of forest_tilemap.bin actually be in it?